src/github/utils/config.ts

```diff
@@ -0,0 +1,108 @@
+import { Value } from "@sinclair/typebox/value";
+import { GitHubContext } from "../github-context";
+import YAML from "yaml";
+import { Config, configSchema } from "../types/config";
+import { expressionRegex } from "../types/plugin";
+import { eventNames } from "../types/webhook-events";
+
+const UBIQUIBOT_CONFIG_FULL_PATH = ".github/ubiquibot-config.yml";
+
+export async function getConfig(context: GitHubContext): Promise<Config | null> {
+  const payload = context.payload;
+  if (!("repository" in payload) || !payload.repository) throw new Error("Repository is not defined");
+
+  const _repoConfig = parseYaml(
+    await download({
+      context,
+      repository: payload.repository.name,
+      owner: payload.repository.owner.login,
+    })
+  );
+  if (!_repoConfig) return null;
+
+  let config: Config;
+  try {
+    config = Value.Decode(configSchema, Value.Default(configSchema, _repoConfig));
+  } catch (error) {
+    console.error("Error decoding config", error);
+    return null;
+  }
+
+  checkPluginChains(config);
+
+  return config;
+}
+
+// eslint-disable-next-line sonarjs/cognitive-complexity
+function checkPluginChains(config: Config) {
```

Comment:
`Refactor this function to reduce its Cognitive Complexity from 58 to the 15 allowed.`
I feel like this rule is too strict, what do you think?

Reply:
Just break the main function apart into smaller functions. I think the rule is fine.